libvcx/src/libindy/utils/wallet.rs

```diff
@@ -153,18 +146,18 @@ pub fn create_wallet(wallet_name: &str, wallet_key: &str, key_derivation: &str,
     }
 }
 
-pub fn create_and_open_as_main_wallet(wallet_name: &str, wallet_key: &str, key_derivation: &str, wallet_type: Option<&str>, storage_config: Option<&str>, storage_creds: Option<&str>) -> VcxResult<WalletHandle> {
+pub fn create_and_open_as_main_wallet(wallet_config: WalletConfig) -> VcxResult<WalletHandle> {
     if settings::indy_mocks_enabled() {
         warn!("open_as_main_wallet ::: Indy mocks enabled, skipping opening main wallet.");
         return Ok(set_wallet_handle(WalletHandle(1)));
     }
-    create_wallet(wallet_name, wallet_key, key_derivation, wallet_type, storage_config, storage_creds)?;
-    open_as_main_wallet(wallet_name, wallet_key, key_derivation, wallet_type, storage_config, storage_creds, None, None)
+    create_wallet(&wallet_config.wallet_name, &wallet_config.wallet_key, &wallet_config.wallet_key_derivation, wallet_config.wallet_type.as_ref().map(String::as_str), wallet_config.storage_config.as_ref().map(String::as_str), wallet_config.storage_credentials.as_ref().map(|s| s.to_string()).as_deref())?;
+    open_as_main_wallet(&wallet_config.wallet_name, &wallet_config.wallet_key, &wallet_config.wallet_key_derivation, wallet_config.wallet_type.as_ref().map(String::as_str), wallet_config.storage_config.as_ref().map(String::as_str), wallet_config.storage_credentials.as_ref().map(|s| s.to_string()).as_deref(), None, None)
```

Comment:
These two should probably take `WalletConfig` as well, right?

Reply:
yup, this change is included in the next PR where I am removing the internal deprecated methods and refactoring internal interfaces.